### PR TITLE
fix: wire or hide beta dead-end controls (Batch H — #1338, #1188, #1191, #1206, #851)

### LINKED ISSUE #851: [Settings][UX] Payment Tracking menu item routes to a placeholder page
## Summary

The visible Settings menu includes a `Payment Tracking` destination, and the URL router exposes `/settings/payment-tracking`, but selecting it only renders a generic `ContentUnavailableView` that says "This page is being built." There is no functional payment/receivables/settings UI behind the advertised menu item.

## Environment/context

Found during Paperclip daily exploratory bug hunt WEI-2350 on `main` / worktree branch `WEI-2350-daily-exploratory-bug-hunt` at commit `beaff3213`.

## Reproduction steps

Static/navigation repro:

1. Open the iOS app settings/user menu.
2. Locate the visible `Payment Tracking` item.
   - Source: `Weird Parts IOS/Weird Parts IOS/Navigation/UserMenuSheet.swift:67-69`
3. Select it, or navigate directly to `/settings/payment-tracking`.
   - Source: `Weird Parts IOS/Weird Parts IOS/Navigation/IOSContentRouter.swift:343-344`
4. Observe the routed view.
   - Source: `Weird Parts IOS/Weird Parts IOS/Features/Settings/SettingsRouter.swift:99-109`

## Expected behavior

A visible `Payment Tracking` menu item should open a usable settings/workflow page for payment tracking, invoices, receivables, or collections, or it should be hidden/disabled until implemented with clear launch criteria.

## Actual behavior

`SettingsRouter` maps `settings-payment-tracking` to:

```swift
case "settings-payment-tracking":
    comingSoonPage("Payment Tracking", icon: "banknote.fill")
```

and `comingSoonPage` renders:

```swift
ContentUnavailableView(title, systemImage: icon, description: Text("This page is being built."))
```

So a searchable, visible production navigation item ends in a placeholder.

## Impact

Medium severity product gap. Office/admin users can discover and open a settings item for a money/receivables workflow, but the app gives no usable action, data, configuration, or next step. Because the menu keywords include `payment`, `tracking`, `invoices`, `receivables`, and `collections`, this looks like a real promised workflow rather than an internal TODO.

## De-duplication

Checked open issues for `Payment Tracking`, `settings-payment-tracking`, `payment tracking`, `receivables`, and `collections`; no duplicate was found.

## Suggested owner

Settings / office-finance owner.

## Acceptance criteria

- Either implement a functional Payment Tracking page, or hide/disable the menu item until the workflow exists.
- If implemented, include the relevant service backing state, empty states, permission/sync scope handling, and tests for routing from `/settings/payment-tracking`.
- If deferred, remove the searchable menu keywords so users are not routed to a placeholder.

### LINKED ISSUE #1188: [Orders][Bug] Parts order bulk action bar only opens coming-soon dead ends
Status: OPEN

## Scanner
Usability Hunter Scanner 3 / feature completeness: visible bulk action buttons do not perform their advertised actions.

## File(s)
- `Weird Parts IOS/Weird Parts IOS/Features/Orders/IOSPartsOrderManagementPage.swift:403-427`

## What is wrong
The parts order management selection bar shows three concrete bulk actions after selecting parts:

- `Move to PO`
- `Change Qty`
- `Remove + Hold`

All three handlers only run `withAnimation { showComingSoon = true }`. They do not move the selected part rows to a purchase order, open a quantity editor, remove/hold parts, or preserve any workflow state beyond showing the generic coming-soon banner.

## Impact
This looks like production functionality but dead-ends during core ordering work. A user can select rows in the order-management queue, choose a bulk action, and get no business result. That blocks efficient PO cleanup and can leave selected ordering work unmanaged.

## Reproduction from code
1. Open `IOSPartsOrderManagementPage` with active supplier rows.
2. Select one or more part rows so `selectionActionBar` appears.
3. Tap any of the three action buttons.
4. Each button only sets `showComingSoon = true`; no order mutation or editor flow runs.

## Expected
Either:
- Wire the actions to real flows (`Move to PO`, `Change Qty`, `Remove + Hold`) with confirmations/validation and service calls, or
- Hide/disable the buttons behind an explicit unavailable state until the feature is implemented, so users do not mistake them for working production actions.

## Related
Parent area: #87 (`[Orders] Purchase Orders — 7-Status Lifecycle, CRM, Delivery Timelines`).

Found during WEI-4107 BugHunter run.

### LINKED ISSUE #1191: [Chat][Bug] Chat exposes dead-end file and thread quick-action controls
Status: OPEN

## Finding
The chat thread UI exposes visible controls that either do nothing or only show a temporary "coming soon" toast, so field users can tap important chat/Q&A actions without any durable result.

## Evidence
- `Weird Parts IOS/Weird Parts IOS/Features/Chat/IOSMessageThreadView.swift:252-257` renders an Attach File button in the composer, but tapping it only sets `showComingSoon = true`.
- `Weird Parts IOS/Weird Parts IOS/Features/Chat/IOSMessageThreadView.swift:120-134` shows the transient message "File attachments coming in a future update" rather than opening a file picker or disabling/hiding the unavailable action.
- `core/Sources/WiredPartCore/Services/ChatService.swift:1141-1157` returns quick actions such as `.addPeople`, `.escalate`, and `.pushBack` for DM/job/QA/RFI/supplier threads.
- `Weird Parts IOS/Weird Parts IOS/Features/Chat/IOSMessageThreadView.swift:619-632` renders those actions as tappable buttons in `ThreadInfoPanel`.
- `Weird Parts IOS/Weird Parts IOS/Features/Chat/IOSMessageThreadView.swift:435-452` only handles `.markResolved`; `.addPeople`, `.escalate`, `.pushBack`, `.approve`, and `.reject` fall through to `break`, producing no UI feedback and no service mutation.

## Reproduction
1. Open any chat thread.
2. Tap the file/document icon in the composer.
3. Observe that the app only shows a temporary coming-soon toast.
4. Open the thread info panel on a job, QA/RFI, DM, or supplier thread where quick actions are listed.
5. Tap Add People, Escalate, or Push Back.

## Expected
Unavailable actions should not be presented as live controls. If a control is visible, tapping it should either perform the action, open the required sheet/confirmation, or show a clear user-facing disabled/unavailable state that explains what is missing.

## Actual
The file attachment button is active but leads to a dead end. Several thread quick-action buttons are rendered from service-provided `availableActions` but the view handler silently ignores them, leaving the user with no result or error.

## Impact
Field chat and Q&A escalation workflows look functional but fail at the point of action. Users may believe they added people, escalated, pushed back, or attached a file when nothing happened, which is especially risky during job support/escalation conversations.

## Suggested fix
Either wire these actions to their service flows and sheets, or filter/hide unavailable actions until implemented. At minimum, replace no-op `break` branches with an explicit disabled state or visible "not available yet" message, and add regression tests ensuring no rendered chat action is silently ignored.

## Dedupe notes
Searched open GitHub issues for `Add People`, `escalate`, `Push Back`, `quick action`, `Attach file`, `File attachments`, and `Coming Soon`. Existing #1178 covers reference-picker load failures and #1188 covers Orders bulk-action coming-soon dead ends; no open issue tracks these chat dead-end controls.

### LINKED ISSUE #1206: [Reports][Bug] Shared Reports exposes a public-report dead end with an empty token
Status: OPEN

## Summary
The Shared Reports tab still shows a "View Public Report" navigation row that opens `IOSPublicReportView(reportToken: "")`. That destination is a stub that always sets `loadError = "Public report sharing is not yet available."`, so users can tap a visible report-sharing control and land in a guaranteed error/dead end.

## Evidence
- `Weird Parts IOS/Weird Parts IOS/Features/Reports/IOSReportsRouter.swift:436-457` renders the visible "View Public Report" row and navigates directly to `IOSPublicReportView(reportToken: "")`.
- `Weird Parts IOS/Weird Parts IOS/Features/Reports/IOSPublicReportView.swift:55-59` ignores the token and always displays "Public report sharing is not yet available."
- `Weird Parts IOS/Weird Parts IOS/Navigation/IOSContentRouter.swift:291-296` already documents that routing `/reports/public` to `IOSPublicReportView` was avoided because the view is a stub; the Shared Reports tab bypasses that safety by instantiating the stub directly.

## Steps to reproduce
1. Open Reports.
2. Switch to Shared Reports.
3. Tap "View Public Report".

## Expected
Users should either get a working public report token entry/viewer flow, or the row should be hidden/disabled behind a clear Coming Soon state until public sharing exists.

## Actual
The row is actionable and always opens a public-report error with an empty token.

## Acceptance criteria
- Shared Reports must not navigate to `IOSPublicReportView(reportToken: "")` while public sharing is unavailable.
- If the feature is intentionally deferred, the row should route to the same Coming Soon placeholder or be removed from active navigation.
- If public report viewing is enabled, provide token input/deep-link handling and load real shared report content.
- Add a regression test that the visible Shared Reports action does not instantiate the public report stub with an empty token.

Found by BugHunter / usability-hunter sweep while working Paperclip WEI-4132.

### LINKED ISSUE #1338: [UX][Bug] Dead-end controls: no-op swipe actions, misrouted quick actions, unreachable pairing flow
Status: IN PROGRESS

Confirmed by the 2026-07-02 deep audit dead-UI scan (adversarially verified; same class as fixed #1256/#1188/#1191). Class PR(s) will fix:

- [ ] **[high]** 'Join Existing Business' onboarding requires a pairing code that no screen can issue or display — either implement host-side code display or gate the flow with clear copy until pairing ships
- [ ] Supplier contact 'Remove' swipe action does nothing (no dialog bound; core delete never called)
- [ ] Warehouse Exec quick actions misroute (wrong tabId format; 'Start Receiving' targets wrong module)
- [ ] Inventory grid 'Transfer' swipe action silent no-op (tab id passed as moduleId; context keys read by nothing)
- [ ] Supplier Bridge settings swallows load errors → misleading 'No Supplier Bridges' empty state
- [ ] Three permanently disabled placeholder buttons with empty actions in sync-related settings pages — remove or wire
- [ ] Write-only/never-read @State cluster (hygiene)

Related open dead-end issues: #851 (payment tracking), #1206 (shared reports token).

### PR DESCRIPTION
**Pull request by xXKillerNoobYT**
Closes #1338
Closes #1188
Closes #1191
Closes #1206
Closes #851

Plan: docs/plans/beta-readiness-issue-resolution-2026-07.md §3 Batch H (dead-end controls umbrella; #851 hide-for-beta decided by owner in #1348 §2).

Rule applied per plan: **wire** a control when the backing service exists and wiring is small, otherwise **hide** it cleanly (no commented-out UI; service code stays).

## What changed

**#1338 — unreachable pairing flow (high)**
- Wired the disabled "Pair New Device" button in Settings → Device Management to `IOSSyncManager.issueShopPairingCode()` (service existed with zero UI callers). It now issues a one-time code and shows it in a sheet with join instructions (`.presentationDetents`, accessibility labels, 44pt target).
- `DevicePairingView` now tells the user where a code comes from (Settings → Device Management → Pair New Device), so "Join Existing Business" is no longer an unexplained dead end.

**#1338 — misrouted / no-op navigation**
- `IOSWarehouseExecPage` quick actions posted tab ids ("movements", "audit") that match no `AppTab.id/path`, and "Start Receiving" targeted the orders module which has no receiving tab. Fixed to `warehouse-movements` / `warehouse-receiving` / `warehouse-audit` on the warehouse module.
- `IOSInventoryGridPage` "Transfer" swipe posted a tab id as `moduleId` ("warehouse-movements") plus context keys read by nothing — a silent no-op. Now posts `moduleId: warehouse`, `tabId: warehouse-movements`.

**#1338 — supplier contact Remove swipe**
- The swipe set `contactToRemove`/`showRemoveContactConfirm` but no dialog was bound and core delete was never called. Added the confirmation dialog and wired it to `PartsService.removeSupplierContact` + contact reload, with errors surfaced.

**#1338 — Supplier Bridge silent load errors**
- `IOSSupplierBridgePage` stored `errorMessage` but never rendered it, masquerading failures as "No Supplier Bridges". Now shows `ErrorStateView` + retry (PR #1343 convention).

**#1338 — three permanently disabled placeholder buttons**
- Device Management "Pair New Device": wired (above).
- Remote Sync "Sync Now" + the fake non-persisting toggle/stepper: removed; page is now honest informational copy.
- Shared Channels "Create Shared Channel": removed until sync infrastructure ships.

**#1338 — write-only state hygiene (partial)**
- Removed the write-only coming-soon/toast state clusters eliminated by these fixes (`showComingSoon` in orders + chat + a never-triggered overlay in `PartsForecastingPage`, fake remote-sync settings state, orphaned selection state). A repo-wide write-only `@State` sweep was not attempted; remaining instances are non-user-facing hygiene.

**#1188 — Parts-order bulk action bar**
- "Move to PO" / "Change Qty" / "Remove + Hold" only opened a coming-soon toast; no backing service flows exist (`updatePOLineItem` is draft-only; no move/hold API). Removed the multi-select checkboxes, action bar, and toast; updated help copy and AI context. Page is read-only for beta as the plan directs; the umbrella tracker for the real flows remains #87.

**#1191 — chat dead-end controls**
- Attach File button now opens a real `.fileImporter` (multi-select, security-scoped copy to temp, size/mime metadata). The attachment pipeline (`sendMessageWithAttachments`, bubble rendering, job-notebook auto-save) already supported `file` type end to end.
- Escalate / Push Back quick actions are wired to new channel-scoped core methods `escalateThreadByChannel` / `pushBackThreadByChannel` (same channel-resolution pattern as `resolveQAThreadByChannel`), with a confirmation dialog for escalate and a required-reason sheet for push back.
- "Add People" (no backing service) is filtered out of rendered quick actions via a supported-actions allowlist; the handler no longer has silent `break` fall-throughs — anything unsupported surfaces a visible message.

**#1206 — Shared Reports public-report dead end**
- Removed the "View Public Report" row that navigated to `IOSPublicReportView(reportToken: "")`, a stub that always errors. The stub view stays for when public sharing ships; `IOSContentRouter` already routes `/reports/public` to a placeholder.

**#851 — Payment Tracking menu item (owner decision #1348: hide for beta)**
- Removed the `UserMenuSheet` item (including its searchable keywords), the `SettingsRouter` case plus the now-unused `comingSoonPage` helper, the `/settings/payment-tracking` route, and the sync-scope entry. The core payment-records service is untouched; the page returns when actually built.

**Tests**
- Core: `ChatServiceTests` — new coverage for `qaThreadId(forChannelId:)`, `escalateThreadByChannel`, `pushBackThreadByChannel` (happy path level transitions + `threadNotFound` on channels without a Q&A thread).
- App target: new `BetaDeadEndControlsRegressionTests` (source-scan style, same pattern as `DashboardScannerEntityContextRegressionTests`) pinning: no public-report stub instantiation, payment-tracking stays hidden, full tab ids in warehouse-exec/inventory-grid navigation, no coming-soon bulk actions, chat actions filtered + fileImporter present, no permanently disabled sync-settings buttons, supplier-bridge error surfacing.

## Test evidence

- `cd core && swift build` → `Build complete!`
- `cd core && swift test --filter ChatServiceTests` → 63 tests passed (includes the 2 new ByChannel tests)
- `cd core && swift test` (full suite) → `Test run with 2260 tests in 68 suites passed after 105.888 seconds.`
- `xcrun swiftc -parse` on all 17 touched app-target files → all parse clean (app files can't be type-checked standalone; diffs visually re-checked)
- App-target regression tests are source-scan based and require an Xcode test run (not executed here; simulator build not run in this environment)

## Notes / gaps

- #1338's "write-only @State cluster" bullet is a hygiene item without a published file list; this PR removes the instances intersecting the fixed dead ends (see above). If the audit's full list surfaces, any remainder is cosmetic (no visible control involved).
- `ThreadInfoPanel` now takes an explicit `actions:` parameter so the desktop/other hosts can pass their own supported set; core still emits `addPeople` for clients that support it.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by xXKillerNoobYT** [comment]
Review finding addressed in 7888bc5b6: `testSharedReportsDoesNotInstantiatePublicReportStub` failed against the committed sources because the #1206 removal NOTE comment in `IOSReportsRouter.swift` quoted the exact UI string the scan forbids. Reworded the comment to 'the public-report viewer row' — the scan still catches any real reintroduction of the row.

Verification: all 28 source-scan assertion executions in `BetaDeadEndControlsRegressionTests` were mechanically re-executed against the branch sources (script replicating each `contains` check): 28/28 pass. The suite itself still has not been run under a simulator — no simulator is available in this environment — so this is mechanical-scan evidence, not an Xcode test run.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

File: Weird Parts IOS/Weird Parts IOS/Features/Settings/IOSDeviceManagementPage.swift [hunk 1]
```diff
@@ -54,16 +78,93 @@ struct IOSDeviceManagementPage: View {
...
+            do {
+                let code = try await appCore.syncManager.issueShopPairingCode()
+                activeSheet = .pairingCode(code)
+            } catch {
+                pairingError = userFriendlyError(error, context: "issue pairing code")
+            }
+            isIssuingCode = false
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (xXKillerNoobYT) [reply Weird Parts IOS/Weird Parts IOS/Features/Settings/IOSDeviceManagementPage.swift:120]: No change needed here. `IOSDeviceManagementPage` is a SwiftUI `View`, which is `@MainActor`-isolated by default in Swift 6 mode (this target builds with SWIFT_VERSION = 6.0). `IOSSyncManager` (the type `appCore.syncManager` refers to) is also explicitly `@MainActor`. Per Swift 6 structured-concurrency rules, a `Task { ... }` created from a `@MainActor`-isolated context inherits that actor unless annotated otherwise, so `issuePairingCode()'s Task body already runs on the MainActor and the `@State` mutations (`activeSheet`, `pairingError`, `isIssuingCode`) are safe. This is well-defined behavior, not undefined UI behavior, so I am leaving the code as-is.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: core/Sources/WiredPartCore/Services/ChatService.swift [hunk 2]
```diff
@@ -1330,6 +1330,38 @@ public final class ChatService: Sendable {
...
+
+    /// Push back the Q&A thread linked to a specific channel.
+    /// Uses `channel_id` to find the correct thread — same pattern as `resolveQAThreadByChannel`.
+    /// Throws `ChatError.threadNotFound` (carrying the channel id) when the channel has no Q&A thread.
+    public func pushBackThreadByChannel(channelId: Int64, pushedBackBy: Int64, reason: String) throws {
+        guard let threadId = try qaThreadId(forChannelId: channelId) else {
+            throw ChatError.threadNotFound(channelId)
+        }
+        try pushBackThread(threadId: threadId, pushedBackBy: pushedBackBy, reason: reason)
+    }
```
[the remark this answers is not in the document]
Comment (xXKillerNoobYT) [reply core/Sources/WiredPartCore/Services/ChatService.swift:1504]: Confirmed and fixed in commit 1b7256f50. Added the same `ServicePermissionGate.requirePermission(..., permissionKey: "moderate_chat")` check to `pushBackThread` that `escalateThread` already has, run in a `db.writer.read` block before the write transaction (matching the existing pattern). This closes the gap where any user reaching the UI could push a Q&A thread back a level without the moderation permission `escalateThread` requires. Verified with `swift build` and `swift test --filter ChatServiceTests` (63/63 passing, including the existing pushBackThread and pushBackThreadByChannel tests, which use an admin user that already holds moderate_chat).

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: Weird Parts IOS/Weird Parts IOS/Features/Chat/IOSMessageThreadView.swift [hunk 3]
```diff
@@ -447,8 +475,75 @@ struct IOSMessageThreadView: View {
...
+                    .appendingPathComponent(UUID().uuidString + "-" + url.lastPathComponent)
+                try FileManager.default.copyItem(at: url, to: tmpURL)
+                let attributes = try? FileManager.default.attributesOfItem(atPath: tmpURL.path)
+                let fileSize = (attributes?[.size] as? NSNumber)?.int64Value
+                pendingAttachments.append(ChatService.PendingAttachment(
+                    type: "file",
+                    filePath: tmpURL.path,
+                    fileName: url.lastPathComponent,
+                    fileSize: fileSize
+                ))
```
[the remark this answers is not in the document]
Comment (xXKillerNoobYT) [reply Weird Parts IOS/Weird Parts IOS/Features/Chat/IOSMessageThreadView.swift:571]: Confirmed and fixed in commit 1b7256f50. `importFiles` now derives `mimeType` via `UTType(filenameExtension: url.pathExtension)?.preferredMIMEType` and passes it into `ChatService.PendingAttachment`, so file attachments populate `mimeType` the same way `fileSize` already does. (Photo attachments still hardcode a `.jpg` temp filename with no real extension to derive a type from, so that path is left as-is — this fix targets the file-import path the comment called out.) Verified with `swift build`; no dedicated attachment-mimeType test exists yet so this was checked by inspection and a clean build.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
